website/static/js/filepage/index.js

```diff
@@ -157,6 +158,74 @@ var FileViewPage = {
         }, 1000);
 
         self.mfrIframeParent = $('#mfrIframeParent');
+
+        $.fn.editable.defaults.mode = 'inline';
+
+        if(self.canEdit) {
+            var $fileName = $('#fileName');
+            var conflict = 'warn';
+            var to = {
+                data: {
+                    nodeId: self.node.id,
+                    path: null,
+                    provider: self.file.provider,
+                }
+            };
+            var from = $.extend(true, {}, to);
+            from.data.path = self.file.path;
+            $fileName.editable({
+                type: 'text',
+                send: 'always',
+                url: waterbutler.moveUrl(),
+                ajaxOptions:{
+                    beforeSend: $osf.setXHRAuthorization,
+                    headers: {
+                        'Content-Type': 'Application/json'
+                    },
+                },
+                params: function(params) {
+                    params = JSON.stringify({
+                        'rename': params.value,
+                        'conflict': conflict,
+                        'source': waterbutler.toJsonBlob(from),
+                        'destination': waterbutler.toJsonBlob(to),
+                    });
+                    return params;
+                },
+                validate: function(value) {
+                    if($.trim(value) === ''){
+                        $osf.growl('Error', 'Name cannot be empty.');
+                        return '   ';
+                    } else if(value.length > 100){
+                        $osf.growl('Error', 'Name should be less than 100 characters');
+                        return '   ';
+                    }
+                },
+                success: function(response, value) {
+                    window.location.reload();
+                },
+                error: function(response) {
+                    var msg = $osf.htmlEscape(response.responseJSON.message);
+                    var msg_long = response.responseJSON.message_long;
+                    if (msg_long) {
+                        $osf.growl('Error', msg_long);
+                        return '      ';
+                    } else if (msg) {
+                        $osf.growl('Error', msg);
+                        return '      ';
+                    } else {
+                        // Log unexpected error with Raven
+                        Raven.captureMessage('Error in renaming file', {
+                            url: waterbutler.moveUrl(),
+                            responseText: response.responseText,
+                            statusText: response.statusText
+                        });
+                        $osf.growl('Error', language.Addons.rename.generalError);
```

Comment:
For cases where the error in renaming was on the server-side and only a general purpose error message is available, growl boxes may still be appropriate; it depends on the error message and consistency with existing UI/ style guides. Some of these growl box uses are more clear cut than others, but take a look.